src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -0,0 +1,397 @@
+
+        ATTRIBUTE_LIST_ID = 0xFFFB
+
+        ATTRIBUTES_TO_CHECK = [
+            RequiredMandatoryAttribute(id=0xFFFD, name="ClusterRevision", validator=check_int_in_range(1, 0xFFFF)),
+            RequiredMandatoryAttribute(id=0xFFFC, name="FeatureMap", validator=check_int_in_range(0, 0xFFFF_FFFF)),
+            RequiredMandatoryAttribute(id=0xFFFB, name="AttributeList",
+                                       validator=check_non_empty_list_of_ints_in_range(0, 0xFFFF_FFFF)),
+            # TODO: Check for EventList
+            # RequiredMandatoryAttribute(id=0xFFFA, name="EventList", validator=check_list_of_ints_in_range(0, 0xFFFF_FFFF)),
+            RequiredMandatoryAttribute(id=0xFFF9, name="AcceptedCommandList",
+                                       validator=check_list_of_ints_in_range(0, 0xFFFF_FFFF)),
+            RequiredMandatoryAttribute(id=0xFFF8, name="GeneratedCommandList",
+                                       validator=check_list_of_ints_in_range(0, 0xFFFF_FFFF)),
+        ]
+
+        success = True
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            for cluster_id, cluster in endpoint.items():
+                for req_attribute in ATTRIBUTES_TO_CHECK:
+                    attribute_string = self.cluster_mapper.get_attribute_string(cluster_id, req_attribute.id)
+
+                    has_attribute = (req_attribute.id in cluster)
+                    location = AttributePathLocation(endpoint_id, cluster_id, req_attribute.id)
+                    logging.info(
+                        f"Checking for mandatory global {attribute_string} on {location.as_cluster_string(self.cluster_mapper)}: {'found' if has_attribute else 'not_found'}")
+
+                    # Check attribute is actually present
+                    if not has_attribute:
+                        self.record_error(self.get_test_name(), location=location,
+                                          problem=f"Did not find mandatory global {attribute_string} on {location.as_cluster_string(self.cluster_mapper)}", spec_location="Global Elements")
+                        success = False
+                        continue
+
+                    # Validate attribute value based on the provided validator.
+                    try:
+                        req_attribute.validator(cluster[req_attribute.id])
+                    except ValueError as e:
+                        self.record_error(self.get_test_name(), location=location,
+                                          problem=f"Failed validation of value on {location.as_string(self.cluster_mapper)}: {str(e)}", spec_location="Global Elements")
+                        success = False
+                        continue
+
+        # Validate presence of claimed attributes
+        if success:
+            logging.info(
+                "Validating that a wildcard read on each cluster provided all attributes claimed in AttributeList mandatory global attribute")
```

Comment:
Followup: we should also validate that all the attributes that actually exist are present in AttributeList.

Reply:
Yes, that reverse side has to be done, but breaks a lot  against too much stuff right now...